services/src/main/java/org/fao/geonet/api/groups/GroupsApi.java

```diff
@@ -541,6 +547,21 @@ public void deleteGroup(
                 ));
             }
 
+            List<Page> staticPages = pageRepository.findAll();
```

Comment:
I would add this method in the repository: 

```java
List<Page> findPageByStatus(Page.PageStatus status);
```
And update the code to the following to at least only process the pages that are assigned to groups:

```java
List<Page> staticPages = pageRepository.findPageByStatus(Page.PageStatus.GROUPS);
List<Page> staticPagesAssignedToGroup =
    staticPages.stream().filter(p ->
            !p.getGroups().stream().filter(g -> g.getId() == groupIdentifier).collect(Collectors.toList()).isEmpty())
        .collect(Collectors.toList());

if (!staticPagesAssignedToGroup.isEmpty()) {
    throw new NotAllowedException(String.format(
        "Group %s is associated with '%s' static page(s). Please remove the static page(s) associated with that group first.",
        group.get().getName(), staticPagesAssignedToGroup.stream().map(p -> p.getLabel()).collect(Collectors.joining())
    ));
}
```

Reply:
@josegar74 

I have added the code. Also the issue you are experiencing "Edit it and change to related to groups and assign it to 2 group --> it doesn't work, it maintains the previous configuration. No error reported." was due to the backend entity relation. It's supposed to be ManyToMany relation. Otherwise, the groupId in the spg_page_group table will be unique. So once one page occupied groupId_1, this groupId_1 will not be available to other pages. Once I have the ManyToMany relation, it does. Here is some example of my local database after the fix:

![image](https://github.com/user-attachments/assets/8f70c134-2204-42c5-b9a0-a07abfb8271b)